static/app/components/avatar/sentryAppAvatar.tsx

```diff
@@ -0,0 +1,33 @@
+import BaseAvatar from 'sentry/components/avatar/baseAvatar';
+import {IconGeneric} from 'sentry/icons';
+import {SentryApp} from 'sentry/types';
+
+type Props = {
+  sentryApp?: SentryApp;
+  isColor?: boolean;
+  isDefault?: boolean;
+} & BaseAvatar['props'];
+
+const SentryAppAvatar = ({isColor = true, sentryApp, isDefault, ...props}: Props) => {
```

Comment:
This style is fine but I'd try to match `SentryAppAvatar` to be the same as `ProjectAvatar` and `OrganizationAvatar`.
```tsx
class SentryAppAvatar extends Component<Props> {
  render() {
    // Render the default if the prop is provided, there is no existing avatar, or it has been reverted to 'default'.
    const avatarDetails = sentryApp?.avatars?.find(({color}) => color === isColor);
    if (isDefault || !avatarDetails || avatarDetails.avatarType === 'default') {
      return (<IconGeneric
        size={`${props.size}`}
        className={props.className}
        data-test-id="default-sentry-app-avatar"
      />);
    }
    return (<BaseAvatar
      {...props}
      type="upload"
      uploadPath="sentry-app-avatar"
      uploadId={avatarDetails?.avatarUuid}
      title={sentryApp?.name}
    />);
  }
}
export default SentryAppAvatar;
```

Reply:
I prefer using React functional components rather than class components. Especially when there is no state managed here

Reply:
@mgaeta Generally if the component has no other method than `render()` we want it to be a function instead of a class. I've refactored `OrganizationAvatar` and `ProjectAvatar` to reflect this.